### PR TITLE
webhooks: Add deprecation notices

### PR DESCRIPTION
**Pull request by ryanslade**
- Add deprecation message to external service config
- Deprecate WebhookLogs field on ExternalService
- Add deprecation notice to batch changes webhook banner
- Update CHANGELOG

Closes https://github.com/sourcegraph/sourcegraph/issues/43569

## Test plan

Only docs and schema changes

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -28,6 +28,8 @@ All notable changes to Sourcegraph are documented in this file.
...
 - Search contexts can now be starred (favorited) in the search context management page. Starred search contexts will appear first in the context dropdown menu next to the search box. [#45230](https://github.com/sourcegraph/sourcegraph/pull/45230)
 - [search.largeFiles](https://docs.sourcegraph.com/admin/config/site_config#search-largeFiles) accepts an optional prefix `!` to negate a pattern. The order of the patterns within search.largeFiles is honored such that the last pattern matching overrides preceding patterns. For patterns that begin with a literal `!` prefix with a backslash, for example, `\!fileNameStartsWithExcl!.txt`. Previously indexed files that become excluded due to this change will remain in the index until the next reindex [#45318](https://github.com/sourcegraph/sourcegraph/pull/45318)
+- New [admin interface](http://localhost:5080/admin/config/webhooks) for webhooks added. Webhooks that were added via code host configuration are [deprecated](https://docs.sourcegraph.com/admin/config/webhooks#deprecation-notice) and will be removed in 4.6.0.
```
Comment (ryanslade) [inline CHANGELOG.md:31]: Our changes are spread over a lot of different PR's so I didn't attached a PR to these lines.

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.00% (0.00 kb) | 0.01% (+0.81 kb) | 0.01% (+0.81 kb) | 0.00% (0) |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/rs/deprecate-webhooks/compare-report.html;tab=live_object) for a full comparison between the commits ef86a71 and 86fc79a or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

File: CHANGELOG.md (hunk 1 above)
Comment (mrnugget) [inline CHANGELOG.md:31]: Why a non-functional link to `localhost:5080`?

How about "Webhooks have been overhauled completely and can now be found under **Site-admin > Webhooks" or something?

File: CHANGELOG.md [hunk 2]
```diff
@@ -28,6 +28,8 @@ All notable changes to Sourcegraph are documented in this file.
...
+- Added support for receiving webhook `push` events from GitHub to respond more quickly when code is committed.
```
Comment (mrnugget) [inline CHANGELOG.md:32]: I'd add _how_ you respond.

**Review by eseliger — APPROVED** [review]

**Review by mrnugget — APPROVED** [review]

**Comment by ryanslade** [comment]
I'm going to wait until Monday before merging this and the side nav change

File: client/web/src/components/externalServices/ExternalServiceWebhook.tsx [hunk 3]
```diff
@@ -59,6 +59,12 @@ export const ExternalServiceWebhook: React.FunctionComponent<React.PropsWithChil
     return (
         <Alert variant="info" className={className}>
             <H3>Batch changes webhooks</H3>
+            <H4>
+                Adding webhooks via code host connections has been{' '}
+                <Link to="/help/admin/config/webhooks#deprecation-notice" target="_blank" rel="noopener noreferrer">
+                    deprecated.
+                </Link>
+            </H4>
```
Comment (LawnGnome) [inline client/web/src/components/externalServices/ExternalServiceWebhook.tsx:67]: I wonder if it's worth us breaking this out to be more fine-grained and only show this message for legacy webhooks. (Not a request for change in this PR, but it's something we @sourcegraph/batchers  should figure out next week.)

**Review by LawnGnome — APPROVED** [review]

**Comment by ryanslade** [comment]
You'll need to merge [this PR](https://github.com/sourcegraph/sourcegraph/pull/45352) first so that the lint check passes.

File: client/web/src/components/externalServices/ExternalServiceWebhook.tsx (hunk 3 above)
Comment (mrnugget) [reply client/web/src/components/externalServices/ExternalServiceWebhook.tsx:67]: Just be mindful that branch cut is today

**Comment by mrnugget** [comment]
> You'll need to merge [this PR](https://github.com/sourcegraph/sourcegraph/pull/45352) first so that the lint check passes.

Done. Rebasing now.

**Pull request merged**